a8cteam51/safety-net
# Use background processes to anonymize many users

### Changes proposed in this Pull Request

* Uses [WP Background Processing](https://github.com/deliciousbrains/wp-background-processing) to anonymize users in the background, which prevents timeout issues when processing large amounts of data
* Moves the admin menu page from Settings to Tools
* Fixes a typo in the CLI class name which was causing an error when running CLI commands

### Notes

Currently it displays the message "Users have been successfully anonymized!" after you click the Anonymize button. This is kind of misleading. The users haven't all been anonymized at this point, the process has only just been scheduled in the background successfully. We should probably change this message.

Ideally, it would be nice to have a status bar that shows what percent of the users have been anonymized, so you can watch it in real time.

### Testing instructions

* Add thousands of users via wp cli: `wp user generate --count=4000`
* Go to Tools > Safety Net and click the Anonymize button to start the process
* When you see the message "Users have been successfully anonymized!" the process has started in the background and users will start to be anonymized
* Go to Users > All Users to check that users have been anonymized. Note: The process could take a few minutes, so you may need to refresh the last page a few times.

Fixes #5
